include/drm/drm_gem.h

```diff
@@ -119,6 +119,11 @@ struct drm_gem_object {
 	 * simply leave it as NULL.
 	 */
 	struct dma_buf_attachment *import_attach;
+
+	/*
+	 * is backed by scattered pages
+	 */
+	bool is_scattered;
```

Comment:
do the CMA APIs require that you pass it a GEM obj wholesale? hm.

I'm very unhappy with this, because I *do* want this driver to eventually go upstream. Do you have plans to eventually complete the unification, or do you imagine you'll be too busy after things are kinda working?